libs/filabridge/include/private/filament/Variant.h

```diff
@@ -98,18 +100,30 @@ namespace filament {
         inline void setFog(bool v) noexcept { set(v, FOG); }
 
         inline constexpr bool isDepthPass() const noexcept {
-            return (key & DEPTH_MASK) == DEPTH_VARIANT;
+            return isValidDepthVariant(key);
+        }
+
+        inline static constexpr bool isValidDepthVariant(uint8_t variantKey) noexcept {
+            // For a variant to be a valid depth variant, all of the bits in DEPTH_MASK must be 0,
+            // except for DEPTH.
+            return (variantKey & DEPTH_MASK) == DEPTH_VARIANT;
         }
 
         static constexpr bool isReserved(uint8_t variantKey) noexcept {
             // reserved variants that should just be skipped
-            return (variantKey & DEPTH_MASK) > DEPTH || (variantKey & 0b010111u) == 0b000100u;
+            // 1. If the DEPTH bit is set, then it must be a valid depth variant. Otherwise, the
+            // variant is reserved.
+            // 2. If SRE is set, either DYN or DIR must also be set (it makes no sense to have
```

Comment:
Be careful with that one actually. Make sure the unlit shadow case (shadow plane for AR) works. It should but...

Reply:
Yup, tested and it still works.